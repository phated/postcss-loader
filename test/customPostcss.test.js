import postcss from "postcss";

import {
  compile,
  getCompiler,
  getErrors,
  getCodeFromBundle,
  getWarnings,
} from "./helpers";

describe('"customPostcss" option', () => {
  it("should work with a custom instance of PostCSS", async () => {
    const spy = jest.fn(postcss);
    const compiler = getCompiler("./css/index.js", {
      // Wrap the spy so it is an instanceof Function
      customPostcss: (...args) => spy(...args),
    });
    const stats = await compile(compiler);
    const codeFromBundle = getCodeFromBundle("style.css", stats);

    expect(spy).toHaveBeenCalledTimes(1);
    expect(codeFromBundle.css).toMatchSnapshot("css");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
    expect(getErrors(stats)).toMatchSnapshot("errors");
  });
});
