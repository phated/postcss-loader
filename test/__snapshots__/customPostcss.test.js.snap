// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"customPostcss" option should work with a custom instance of PostCSS: css 1`] = `
"a {
  color: black;
}

a {
  color: red;
}

a {
  color: green;
}

a {
  color: blue;
}

.class {
  -x-border-color: blue blue *;
  -x-color: * #fafafa;
}

.class-foo {
  -z-border-color: blue blue *;
  -z-color: * #fafafa;
}

.phone {
  &_title {
    width: 500px;

    @media (max-width: 500px) {
      width: auto;
    }

    body.is_dark & {
      color: white;
    }
  }

  img {
    display: block;
  }
}
"
`;

exports[`"customPostcss" option should work with a custom instance of PostCSS: errors 1`] = `Array []`;

exports[`"customPostcss" option should work with a custom instance of PostCSS: warnings 1`] = `Array []`;
